projects/packages/waf/actions.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * Action Hooks for Jetpack WAF module.
+ *
+ * @package automattic/jetpack-waf
+ */
+
+namespace Automattic\Jetpack\Waf;
+
+// We don't want to be anything in here outside WP context.
+if ( ! function_exists( 'add_action' ) ) {
+	return;
+}
+
+/**
+ * Runs the WAF in the WP context.
+ *
+ * @return void
+ */
+add_action(
+	'plugin_loaded',
+	function () {
+		$allowed_modes = array(
+			'normal',
+			'silent',
+		);
+
+		$mode_option = get_option( 'jetpack_waf_mode' );
+
+		if ( ! in_array( $mode_option, $allowed_modes, true ) ) {
+			return;
+		}
+
+		define( 'JETPACK_WAF_MODE', $mode_option );
```

Comment:
I think we'll need to check if the constant is already defined before we set it, to avoid warnings like this one:

```
PHP Notice:  Constant JETPACK_WAF_MODE already defined in /wp-content/plugins/jetpack-dev/vendor/automattic/jetpack-waf/actions.php on line 34
```

Reply:
Added in a way that also makes sure that the mode is one of the allowed ones if the constant is defined beforehand.